src/gui/MainWindow.cpp

```diff
@@ -484,6 +484,15 @@ void MainWindow::initGui()
 
     if(mConfig->useTrayIcon()) {
 	    connect(mTrayIcon, &TrayIcon::showEditorTriggered, [this](){ mVisibilityHandler->enforceVisible(); });
+		connect(mTrayIcon, &TrayIcon::leftClickActionTriggered, [this](){
```

Comment:
Shouldn't we keep this logic in the TrayIcon class? It has all it requires to do the job and the outside world doesn't even need to know about default actions?

Reply:
@DamirPorobic The `TrayIcon` has access to the config, via the `KsnipConfigProvider::instance()` singleton, but it does not have access to `MainToolBar` and `WidgetVisibilityHandler`.

Reply:
But why would it need access to does two? I don't have the code in front of me but I think that the TrayIcon can already trigger new screenshots of all supported types, can't we utilize this? Based on the selected default action type you just select a matching action from all supported actions in the TrayIcon and trigger it, the TrayIcon sends a signal to the MainWindow and the MainWindow does its stuff.

Reply:
OK, understood